app/assets/stylesheets/variables/_utilities.scss

```diff
@@ -0,0 +1,96 @@
+@use 'sass:map';
+
+$utilities-package: map.remove(
+  $utilities-package,
+  'align-self',
+  'add-aspect',
+  'bottom',
+  'box-shadow',
```

Comment:
Just to check my understanding:

So today, if I wanted to use [`shadow-2`](https://designsystem.digital.gov/utilities/shadow/#:~:text=0%2C%200%2C%200.1)-,.shadow%2D2,-0%204px%208px), I could just use it.

But with these changes, it wouldn't work automaticall, so I'd have to know to edit this file and then remove this line here so it gets added in to our CSS, right?

Reply:
Yeah, that's right.

A few thoughts on it:

- It would at least be obvious that it isn't working, since it wouldn't apply the styled effect when previewed
- I grant that it could be confusing to try to debug to know what to do next
   - One of the thoughts I had was using [ERBLint DeprecatedClass](https://github.com/Shopify/erb-lint/#deprecatedclasses) opposite its intended usage, as a way to inform someone how to enable a currently-disabled class. I have a feeling most people wouldn't see this warning until long after it's relevant though.
- I've been thinking we should consider relying a bit less on utility classes for styling core components. Ad hoc styles like margins and padding seem fine enough, but in a case like [the `nav_branded.html.erb` changes](https://github.com/18F/identity-idp/pull/7262/commits/e0c52b826e1c3bcdd85ea4d9b057d214fc9d94aa#diff-5ee783c4d56440497e845f79ea0053bb7bcebafdead76cd6186db5d2442127e8), we should probably consider having a "logo(s)" component, and applying the styles there. We can still use design system utility Sass mixins (e.g. `@include u-shadow`) even if disabled here.

Reply:
I like that approach!

Reply:
> * It would at least be obvious that it isn't working, since it wouldn't apply the styled effect when previewed

Maybe it's because I've done more CSS in our Jekyll apps than the IDP, but sometimes things don't auto-refresh for me on the first time and I spend a few refreshes guessing if I made a typo or not (or equivalent)

I'm a little skeptical of the workflow implications for this, but willing to give it a shot and check back in ~3 months at a future eng huddle (so we can make sure new hires, teammates aren't confused by this)

Another thought, our goal is to reduce the size of the resulting production CSS. Since we mostly avoid interpolated CSS classes, what if we loaded everything in development mode, but for production, we wrote a little filter that strips out CSS classes not found in our source  code? (at least from the USWDS styles). For the time being, all our component HTML is defined in this repo, so it means we'd only have one place to look?

Reply:
Yeah, our main stylesheet is our largest single asset for the critical paths (by more than 2x the next largest), so I've been trying to think of ways to reign that back in.

I think [USWDS v3's modular structure](https://designsystem.digital.gov/whats-new/updates/2022/04/28/introducing-uswds-3-0/) will help here, since I think it'll hopefully be a little more developer friendly to opt-in _components_ as opposed to utility classes. But we're currently blocked on upgrading since they've dropped Internet Explorer support. 

I like the idea of removing CSS classes which are unused based on our source code, but I think that'll be really technically challenging to get right, especially if it's only applied in certain environments (though I'd understand why we wouldn't want to do this kind of processing in local development).

As of now, I'm leaning towards one of...

- Keep this work, but limit it to utility classes that we have relatively high certainty we wouldn't ever want to use
- Shelve it, and consider revisiting the component opt-in with USWDS v3 once we drop support for Internet Explorer and can upgrade